shell/platform/windows/win32_window.cc

```diff
@@ -195,19 +202,21 @@ Win32Window::MessageHandler(HWND hwnd,
       case WM_SYSDEADCHAR:
       case WM_CHAR:
       case WM_SYSCHAR: {
-        char32_t code_point = static_cast<char32_t>(wparam);
-        static char32_t s_pending_lead_surrogate = 0;
-        // If code_point is LeadSurrogate, save to combine to potentially form
-        // a complex Unicode character.
-        if ((code_point & 0xFFFFFC00) == 0xD800) {
-          s_pending_lead_surrogate = code_point;
-        } else if (s_pending_lead_surrogate != 0 &&
-                   (code_point & 0xFFFFFC00) == 0xDC00) {
-          // Merge TrailSurrogate and LeadSurrogate.
-          code_point = 0x10000 +
-                       ((s_pending_lead_surrogate & 0x000003FF) << 10) +
-                       (code_point & 0x3FF);
-          s_pending_lead_surrogate = 0;
+        static wchar_t s_pending_high_surrogate = 0;
+
+        wchar_t character = static_cast<wchar_t>(wparam);
+        std::u16string text({character});
+        char32_t code_point = character;
+        if (IS_HIGH_SURROGATE(character)) {
```

Comment:
I don't see where this macro is defined.

Reply:
Oh. Nevermind. Its a Windows thing https://docs.microsoft.com/en-us/windows/win32/api/winnls/nf-winnls-is_high_surrogate.

Reply:
Yep, I happened across a mention of it in the docs while looking up something else for this PR, and it seemed like a good opportunity to replace some hand-rolled code with something that was already available.